data-prepper-plugins/rss-source/src/main/java/org/opensearch/dataprepper/plugins/source/rss/RSSSourceConfig.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.rss;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.NotBlank;
+
+import java.time.Duration;
+
+public class RSSSourceConfig {
+
+    static final Duration DEFAULT_POLLING_FREQUENCY = Duration.ofMinutes(5);
+
+    @JsonProperty("url")
+    @NotBlank(message = "RSS Feed URL cannot be null or empty")
+    private String url;
+
+    @JsonProperty("pollingFrequency")
+    private Duration pollingFrequency = DEFAULT_POLLING_FREQUENCY;
```

Comment:
I understand the definition difference. However, the reason it's being called `pollingFrequency` is due to retrieving the RSS feed URL every n minutes, or how frequently we retrieve the refreshed feed. I'd avoid calling it Period related since period in Java usually refers to date-based values. We can sync-up if you still think it should be renamed.

Reply:
Discussed offline. The format will be a notation string, e.g.

```
pollingFrequency: 5m
```

Reply:
This will be called out in the RSS Source documentation. Duration values will support ISO-8601 format so `PT5M` for 5 minutes as well as notation Strings for seconds and milliseconds (e.g. 60s or 3600ms)